Source/Diagnostics/ReducedDiags/ParticleNumber.cpp

```diff
@@ -0,0 +1,103 @@
+/* Copyright 2019-2020 Neil Zaim, Yinjian Zhao
+ *
+ * This file is part of WarpX.
+ *
+ * License: BSD-3-Clause-LBNL
+ */
+
+#include "ParticleNumber.H"
+#include "WarpX.H"
+
+// constructor
+ParticleNumber::ParticleNumber (std::string rd_name)
+: ReducedDiags{rd_name}
+{
+    // get WarpX class object
+    auto & warpx = WarpX::GetInstance();
+
+    // get MultiParticleContainer class object
+    auto & mypc = warpx.GetPartContainer();
```

Comment:
Can this be `const`?
```suggestion
    const auto & mypc = warpx.GetPartContainer();
```